Bottom: aa8f70a5a4e7bc141fd2ab499718aa266dfa2afa
Top:    736ecf705dc809ec3cb3b01440b26315113e5ace
Author: Jon Smirl <jonsmirl@gmail.com>
Date:   2014-07-14 21:06:01 -0400

Refresh of audio

---

diff --git a/drivers/dma/sun4i-dma.c b/drivers/dma/sun4i-dma.c
index ca33f8c..5721329 100644
--- a/drivers/dma/sun4i-dma.c
+++ b/drivers/dma/sun4i-dma.c
@@ -510,6 +510,7 @@ static struct sun4i_dma_contract *generate_dma_contract(void)
 		return NULL;
 
 	INIT_LIST_HEAD(&contract->demands);
+printk("JDS DMA demands init next %p prev %p\n", contract->demands.next, contract->demands.prev);
 	INIT_LIST_HEAD(&contract->completed_demands);
 
 	return contract;
@@ -939,6 +940,8 @@ static irqreturn_t sun4i_dma_interrupt(int irq, void *dev_id)
 			spin_lock(&vchan->vc.lock);
 			if (contract->cyclic) {
 				/* move promise to back of list */
+
+printk("JDS DMA demands next %p prev %p\n", contract->demands.next, contract->demands.prev);
 				list_del_init(&vchan->processing->list);
 				list_add_tail(&vchan->processing->list, &contract->demands);
 				vchan->processing = NULL;
